plugins/serialization/src/main/java/org/qbicc/plugin/serialization/BuildtimeHeap.java

```diff
@@ -82,12 +93,29 @@ public synchronized Data serializeStringLiteral(String value) {
         return sl;
     }
 
+    public synchronized Data serializeClassObject(LoadedTypeDefinition type) {
+        if (classObjects.containsKey(type)) {
+            return classObjects.get(type);
+        }
+
+        LoadedTypeDefinition jlc = ctxt.getBootstrapClassContext().findDefinedType("java/lang/Class").load();
+        Data cl = serializeClassObjectImpl(jlc, type);
+        classObjects.put(type, cl);
+        return cl;
+    }
+
     public synchronized Data serializeObject(Object obj) {
         if (objects.containsKey(obj)) {
             return objects.get(obj);
         }
 
         Class<?> cls = obj.getClass();
+        // java.lang.Class instances are their own special thing; can't just reflectively write them.
+        if (obj instanceof Class<?>) {
+            LoadedTypeDefinition ltd = ctxt.getBootstrapClassContext().findDefinedType(cls.getName().replace('.', '/')).load();
```

Comment:
Will this work for non-bootstrap classes?  I think this will work for now but once the interpreter is up and running, we may need to revisit this as there may be other ClassContexts.

Reply:
Expect it won't work as is.  We'll need a way to map from the ClassLoader of the Class to the ClassContext we are supposed to use to load the qbicc type.  Assuming this will be possible, but we don't have it yet.

Reply:
The interpreter has an API for loading classes, so this would be how it is done once that is ready.